airflow/providers/elasticsearch/hooks/elasticsearch.py

```diff
@@ -34,6 +34,9 @@
 
     from airflow.models.connection import Connection as AirflowConnection
 
+if "pytest" in sys.modules:
+    import elasticsearch  # noqa: F401
```

Comment:
This sounds very strange and I would never expect to check in "prod" code whether pytest is in modules. That is very suspicious - what happens if someone does have pytest in their dependencies (for whatever reason?) I do not understand why this one would be needed here. It's also under TYPE_CHECKIN Why do you need to check for pytest here?

IMHO. Way better solution will be to make "from elasticsearch import ElasticSearch"  local in all the methods that need it and leave import elasticsearch unconditionally in TYPE_CHECKING.

Reply:
Thanks. Yes indeed this doesn't look good. I was doing it originally for unittest patch to be easier. 

I've removed it and rewrite the test to patch on `elasticsearch.Elasticsearch` directly.